package-dev/Runtime/SentryInitialization.cs

```diff
@@ -142,12 +178,20 @@ private Il2CppMethods _il2CppMethods
         private static void Il2CppNativeStackTraceShim(IntPtr exc, out IntPtr addresses, out int numFrames, out string? imageUUID, out string? imageName)
         {
             imageName = null;
-            // Unity 2020 does not *return* a newly allocated string as out-parameter, but rather expects a pre-allocated buffer it writes into.
-            // That buffer needs to have space for the hex-encoded uuid (32) plus terminating nul-byte.
-            var uuidBuffer = new char[32 + 1];
+            // Unity 2020 does not *return* a newly allocated string as out-parameter,
+            // but rather expects a pre-allocated buffer it writes into.
+            // That buffer needs to have space for either:
+            // - A hex-encoded `LC_UUID` on MacOS (32)
+            // - A hex-encoded GUID + Age on Windows (40)
+            // - A hex-encoded `NT_GNU_BUILD_ID` on ELF (Android/Linux) (40)
+            // plus a terminating nul-byte.
+            var uuidBuffer = new char[40 + 1];
```

Comment:
`char` is 2 bytes in c# but we're passing to C `char` in `const Il2CppException * ex, uintptr_t** addresses, int* numFrames, char* imageUUID)`, We should be using `byte[]` here and in the other call to `il2cpp_native_stack_trace` instead. If we do, we can than make `SanitizeDebugId` also take the same `byte` array, making it possible to efficiently swap the bytes as needed.

Conversion to string can happen when needed.

Reply:
@Swatinem this might be why it's not going out of bounds previously when giving it a buffer of 32+1.

Reply:
Indeed, internally `Il2CppChar` is defined as u16 like this:

```
#if _MSC_VER
typedef wchar_t Il2CppChar;
#elif __has_feature(cxx_unicode_literals)
typedef char16_t Il2CppChar;
#else
typedef uint16_t Il2CppChar;
#endif
```

So I’m wondering how this could have ever worked.